### PR TITLE
fix: Fix upgrade name causing panic on migrations

### PR DESCRIPTION
**Pull request by amaurym**
## Description

Sorry, I forgot to update this in #403 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/stable_appconfig.go [hunk 1]
```diff
@@ -41,7 +41,8 @@ func setCustomKVStoreKeys() []string {
 }
 
 func (app *RegenApp) registerUpgradeHandlers() {
-	app.UpgradeKeeper.SetUpgradeHandler("v0.43.0-rc0-upgrade", func(ctx sdk.Context, plan upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
+	upgradeName := "v0.43.0-rc0-upgrade"
```
Comment (amaurym) [inline app/stable_appconfig.go:44]:
```suggestion
	// This is the upgrade plan name we used in the gov proposal.
	upgradeName := "v0.43.0-rc0-upgrade"
```

File: app/stable_appconfig.go [hunk 2]
```diff
@@ -41,7 +41,10 @@ func setCustomKVStoreKeys() []string {
 }
 
 func (app *RegenApp) registerUpgradeHandlers() {
-	app.UpgradeKeeper.SetUpgradeHandler("v0.43.0-rc0-upgrade", func(ctx sdk.Context, plan upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
+	// This is the upgrade plan name we used in the gov proposal.
+	// This is the upgrade plan name we used in the gov proposal.
```
Comment (amaurym) [inline app/stable_appconfig.go:45]:
```suggestion
```

File: app/stable_appconfig.go [hunk 3]
```diff
@@ -72,7 +74,7 @@ func (app *RegenApp) registerUpgradeHandlers() {
 		panic(err)
 	}
 
-	if upgradeInfo.Name == "v0.43.0-beta1-upgrade" && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+	if upgradeInfo.Name == upgradeName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
```
Comment (robert-zaremba) [inline app/stable_appconfig.go:77]: if we will have `v0.43.0-rc1`, we will need to update this again, right?
How about doing this change:
```
if stirngs.HasPrefix(upgradeInfo.Name, "v0.43.0") ...
```

**Review by robert-zaremba — APPROVED** [review]
LGTM, left one comment which could be an improvement for next upgrade.

File: app/stable_appconfig.go (hunk 3 above)
Comment (anilCSE) [reply app/stable_appconfig.go:77]: This is just the upgrade handler name, it may not refer to the original version.

**Review by anilCSE — APPROVED** [review]
ACK

**Pull request merged**